lib/bolt/catalog.rb

```diff
@@ -51,6 +51,17 @@ def generate_ast(code)
       end
     end
 
+    def setup_inventory(inventory)
```

Comment:
This whole function feels messy. I don't have a better idea at the moment though.